### DISCUSSION TITLE
Add files.upload filter

### DISCUSSION THREAD

**Discussion by benosman** [directus/directus#16584] [Feature Requests]
There is a files.upload action but not a filter. 

It would be handy if the filter was there to give the chance to block / amend a file before the database record is created rather than the current situation where we need to use the files.upload action and delete the recently created file record.

I have a pull request ready if this would be considered. 

Related discussions:
https://github.com/directus/directus/discussions/11733
https://github.com/directus/directus/discussions/15743
https://github.com/directus/directus/discussions/5162

**Comment by alexpyattaev** [comment]
I second this motion. We have a data-heavy application so having a way to ensure quota *before* the files drop on the server would be nice.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by dnl13** [comment]
This topic needs much more attention

**Reply by br41nslug** [reply]
please do elaborate 😄 "needs more attention" is not really something actionable on an old closed feature request.
